### PR TITLE
Alter Xeno Evolution: Larva -> Drone Mandatory

### PR DESCRIPTION
**Pull request by Venuska1117**
# About the pull request
This PR make drone caste "Basic", it  is made based  on this post made on Ideaguys Page:
https://forum.cm-ss13.com/t/alter-xeno-evolution-larva-drone-mandatory/8075

# Explain why it's good for the game

Meke drone "Basic" for other castes, with bring some positives:
- People will be more active as drone, give them something to do when they wait for other castes.
- Good choice for new players, players will get used to playing xenos more and survive for longer time, it will prevent early death caused by choosing runner caste and dying ( Xeno mains know, sometimes you will sit 45 minutes in queue)

Negatives?
-  will create confusion to old players for short time.
- some people dislike changes, even if change don't change anythin essential (other than default evolve tree) it WILL be hated.

Only issue with this change, can be "caste swapping" in combat (like drone -> Runner), if it will be too much of issue, i can add timer (like 30-60 seconds) where you need to "mature" before you can use your abilities (like how lesser drone have), but it require testing first before making any changes.
 
# Testing Photographs and Procedure

Fully unlocked Evolution choices:
image

# Changelog
:cl: Venuska1117
balance: Larva evolve only to Drone, Now Drones are "Basic" Caste, they now can evolve to: Runner, Sentiel, Defender, Queen, Burrower and Carrier.
balance: Evolving from: Larva to drone and drone to other T1 castes cost 30 evolution.
/:cl:

### CODE REVIEW

**Comment by InsaneRed** [comment]
this is a balance change not a code improvement change

**Comment by Venuska1117** [comment]
> this is a balance change not a code improvement change

woops, sorry let me fix that

**Comment by InsaneRed** [comment]
1.) Just because people evolve into drones doesnt mean people will go around weeding.
2.) This effects late game balance as you're still required evo points to evolve from drone....to anything? i dont know how you coded this. so later rounds would get gimped
3.) Bonus evo was already added so people can become a drone and then go any caste they want.

**Comment by HeresKozmos** [comment]
Finally, a good PR.

**Comment by cuberound** [comment]
based on the  Ideaguys Page you are trying to implement  the drone should gain evo when queen is not in oviposition or (better) have separete points for evolving from drone to other t1 and from drone to t2 and make it so that you gain the t1 evo points even with queen out of oviposition

**Comment by InsaneRed** [comment]
If you make it so its 60 evolution to larva -> to drone, how are you planning on solving the issue of drone to any other tier 1? is it going to be free evolution? That'd be causing time issues on how fast you can join the front if it isnt. 

From a marine prespective, you'd be allowing newly born alrvas to make weeds walls and more reinforcements while they wait for the allocated x free evolution (before freely evolving to any other "combat" caste.) which would double the efficiency of any new larva born into the hive, how do you intend to solve this?

The whole larva > drone > tier 1 > tier 2 > tier 3 time gets way longer considering all this, AND the fact that tier 3's are unlocked at minute 20 which would mean in some cases you wouldnt even be getting tier 3 despite marines having landed, due to some late queen spawns, or maybe when a queen isnt chosen round start so one has to evolve then get on ovi how do you plan on solving that?

The biggest issue with this pr is the fact that its un-required, we already have a perfectly working caste-system that saves evo for people who play drone in the sense of locked evo timers and reserve evolution points

**Comment by ihatethisengine** [comment]
I'd like to keep my free will to play or not to play drone.

**Comment by Steelpoint** [comment]
The addition of a forced drone evo should not prevent xenos getting t3s before marines arrive, especially if Drone to T1 combat caste has a lower evo time

File: code/modules/mob/living/carbon/xenomorph/castes/Drone.dm [hunk 1]
```diff
@@ -20,7 +20,7 @@
 	build_time_mult = BUILD_TIME_MULT_BUILDER
 
 	caste_desc = "A builder of hives. Only drones may evolve into Queens."
-	evolves_to = list(XENO_CASTE_QUEEN, XENO_CASTE_BURROWER, XENO_CASTE_CARRIER, XENO_CASTE_HIVELORD) //Add more here separated by commas
+	evolves_to = list(XENO_CASTE_RUNNER, XENO_CASTE_SENTINEL, XENO_CASTE_DEFENDER, XENO_CASTE_QUEEN, XENO_CASTE_BURROWER, XENO_CASTE_CARRIER, XENO_CASTE_HIVELORD) //Add more here separated by commas
 	deevolves_to = list("Larva")
```
Comment (Drulikar) [inline code/modules/mob/living/carbon/xenomorph/castes/Drone.dm:24]: These additional castes then should have their deevolves_to setting changed from larva to drone then. Additionally, the cost of evolution to previous T1s then need to be compensated for the cost of evolving to drone. See `/datum/caste_datum/New()` for how this previously was handled.

**Review by Drulikar — CHANGES_REQUESTED** [review]
For this to be approved, I do not want the timing to ultimately change for any caste. So that is to say, the speed to become drone should be the same as it was, and T1s that evolved from T0 are compensated for having to go through T1 twice.

File: code/modules/mob/living/carbon/xenomorph/castes/Drone.dm (hunk 1 above)
Comment (Drulikar) [reply code/modules/mob/living/carbon/xenomorph/castes/Drone.dm:24]: Nevermind the larva bit especially if we're tweaking the evolution times.

**Comment by Steelpoint** [comment]
I sincearly believe now that the best way to make this change work smoothly, is to make it that once you have evolved past Drone, you can not de-evolve back into a Drone (with exception for support xenos). Basically you have to become a Drone, from Larva, but once you are past being a Drone you return back to being a Larva if you de-evolve. 

This very much eliminates all possible forms of abuse that I can think of, and is the most elegant solution to said issues without requiring a massive host of further code changes.

**Comment by Venuska1117** [comment]
After many takes to this issue, i concluded than i am not capable enough to make this to reality, if we want make everything work as it work, it require to rework evolution.dm system with i am not good at, of course there would be many different ways to solve this issue, but my lack of deeper BYOND coding knowledge make me unable to do that, i hope someone else will get courage to do it- or if i learn BYOND coding better (i only know basics of python code for now).

**Comment by Venuska1117** [comment]
After many takes to this issue, i concluded than i am not capable enough to make this to reality, if we want make everything work as it work, it require to rework evolution.dm system with i am not good at, of course there would be many different ways to solve this issue, but my lack of deeper BYOND coding knowledge make me unable to do that, i hope someone else will get courage to do it- or if i learn BYOND coding better (i only know basics of python code for now).

File: code/modules/mob/living/carbon/xenomorph/castes/Drone.dm (hunk 1 above)
Comment (Venuska1117) [reply code/modules/mob/living/carbon/xenomorph/castes/Drone.dm:24]: whole system got changed, please check code again.

**Comment by Venuska1117** [comment]
I will type list of change here, later i will update description (i need go to work soon and don't want to forget):
larva -> drone (cost 30 evo points, recommended wait for 60evo threshold) 
drone -> runner, defender and sentinel (require 30 evo points, can be carried from drone stage (if you hit 60 evo threshold))
- Can evolve to other T1 as drone when queen die, but you cannot evolve to T2
- De-evolve as runner, defender and sentinel make you larva instead of drone (to prevent abusement)

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Pull request closed without merging**